### PR TITLE
Feature/externalsync

### PR DESCRIPTION
**Pull request by raabuchanan**
Adds functionality to sync realsense with external signal

### CODE REVIEW

File: realsense2_camera/include/constants.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ namespace realsense2_camera
     const bool ALIGN_DEPTH    = false;
     const bool POINTCLOUD     = false;
     const bool SYNC_FRAMES    = false;
+    const bool FORCE_MAVROS_TRIGGERING = false;
```
Comment (YoshuaNava) [inline realsense2_camera/include/constants.h]: How does MAVROS relate to general use cases for external syncing?

Comment (marco-tranzatto) [reply realsense2_camera/include/constants.h]: This is merged in another branch that is NOT master. Used for testing on Bear. No worries ;)

**Comment by marco-tranzatto** [comment]
This is merged in another branch that is NOT master. Used for testing on Bear. No worries ;)

File: realsense2_camera/src/base_realsense_node.cpp [hunk 2]
```diff
@@ -1378,11 +1408,21 @@ void BaseRealSenseNode::publishFrame(rs2::frame f, const ros::Time& t,
         img->step = width * bpp;
         img->header.frame_id = optical_frame_id.at(stream);
         img->header.stamp = t;
-        img->header.seq = seq[stream];
+        img->header.seq = image_counter_;//f.get_frame_number();
```
Comment (lorenwel) [inline realsense2_camera/src/base_realsense_node.cpp]: We should be careful setting the outgoing message sequence number. [ROS is known to overwrite this](https://answers.ros.org/question/55126/why-does-ros-overwrite-my-sequence-number/). Did you verify that this sequence number is actually the one being published in the outgoing message?

File: realsense2_camera/cfg/base_d400_params.py [hunk 3]
```diff
@@ -7,7 +7,7 @@ def add_base_params(gen, prefix):
   # Level should match the order of appearance in base_depth_param of base_realsense_node.h
   #             Name                                               Type    Level Description                  Default    Min     Max
   gen.add(str(prefix) + "depth_gain",                              int_t,    1,  "Gain",                      16,        16,     248)
-  gen.add(str(prefix) + "depth_enable_auto_exposure",              bool_t,   2,  "Enable Auto Exposure",      True)
+  gen.add(str(prefix) + "depth_enable_auto_exposure",              bool_t,   2,  "Enable Auto Exposure",      False)
```
Comment (lorenwel) [inline realsense2_camera/cfg/base_d400_params.py]: Will this be a permanent default value change or is this just for testing?

File: realsense2_camera/cfg/rs435_params.cfg [hunk 4]
```diff
@@ -17,7 +17,7 @@ gen.add("rs435_depth_laser_power",               double_t,  10,    "Laser Power"
 emitter_enabled_enum = gen.enum([gen.const("Off",  int_t,  0,  "Off"),
                                  gen.const("On",   int_t,  1,  "On"),
                                  gen.const("Auto", int_t,  2,  "Auto")], "Depth Emitter")
-gen.add("rs435_depth_emitter_enabled",           int_t,     11,    "Depth Emitter Enabled",     1,         0,      2, edit_method=emitter_enabled_enum)
+gen.add("rs435_depth_emitter_enabled",           int_t,     11,    "Depth Emitter Enabled",     0,         0,      2, edit_method=emitter_enabled_enum)
```
Comment (lorenwel) [inline realsense2_camera/cfg/rs435_params.cfg]: As above, permanent, or for testing?

**Comment by YoshuaNava** [comment]
> This is merged in another branch that is NOT master. Used for testing on Bear. No worries ;)

I saw that :) Thanks.

**Pull request merged**